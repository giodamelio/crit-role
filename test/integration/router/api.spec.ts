import supertest, { SuperTest, Test } from 'supertest';

import db from '../../../src/database';

// Test subject
import setupServer from '../../../src/index';

describe('Router', () => {
  describe('api', () => {
    let request: SuperTest<Test>;
    beforeAll(async () => {
      const server = await setupServer();
      request = supertest.agent(server.callback());
    });

    afterAll(async () => {
      await db.destroy();
    });

    it('should succesfully ping', () => {
      expect.assertions(0);

      return (
        request
          // Pong is important, ok?
          .get('/api/ping')
          .expect(200)
          .expect({ response: 'pong' })
      );
    });

    it('should list the people', () => {
      expect.assertions(1);

      return request
        .get('/api/people')
        .expect(200)
        .then((res) => {
          expect(res.body.people).toMatchSnapshot();
        });
    });
  });
});
