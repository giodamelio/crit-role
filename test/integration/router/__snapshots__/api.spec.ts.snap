// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router api should list the people 1`] = `
Array [
  Object {
    "id": "ashley-johnson",
    "name": "Ashley Johnson",
  },
  Object {
    "id": "laura-bailey",
    "name": "Laura Bailey",
  },
  Object {
    "id": "liam-obrien",
    "name": "Liam O'Brien",
  },
  Object {
    "id": "marisha-ray",
    "name": "Marisha Ray",
  },
  Object {
    "id": "matthew-mercer",
    "name": "Matthew Mercer",
  },
  Object {
    "id": "sam-riegel",
    "name": "Sam Riegel",
  },
  Object {
    "id": "taliesin-jaffe",
    "name": "Taliesin Jaffe",
  },
  Object {
    "id": "travis-willingham",
    "name": "Travis Willingham",
  },
]
`;
